### DISCUSSION TITLE
Noise calculations for circuits containing noisy networks: Questions and resources

### DISCUSSION THREAD

**Discussion by Stijnvdv** [scikit-rf/scikit-rf#995] [General]
Hi!

Recently I have been trying to use SKRF for system level modelling of an RF link. When limiting the lineup to a cascade of twoports the noise calculations line up. Now when I wanted to move to more then just a cascaded of twoports but actually use `skrf.Circuit` class to build a netlist I noticed there is no noise calculation for circuits. When researching the topic I realized using noise waves it can be done, in a similar fashion to the actual S-parameter calculations in `skrf.Circuit`. Seemingly this [QUCS paper](https://qucs.github.io/docs/technical/technical.pdf) has everything I need.

I just wanted to get some input from you guys on plans of attack and maybe some good resources about the topic. I also have some open questions that some of you can resolve:

- The `skrf.Network.n` uses the "A" cascade representation for the noise. After some digging I have not found any resources about the construction of this matrix. Someone have a good reference?
- How does one convert the cascade representation to noise waves? And would this be a good approach?
- Does switching the way noise is handled in `skrf.Network` make sense? Switching away from cascade representation?

When pointed into the right direction I think I can manage to make it work and implement it.
Thanks in advance! 73

**Comment by jhillairet** [comment]
> When limiting the lineup to a cascade of twoports the noise calculations line up. Now when I wanted to move to more then just a cascaded of twoports but actually use skrf.Circuit class to build a netlist I noticed there is no noise calculation for circuits.

Indeed, `Circuit` does not support (yet?) noise data.

Note that if you only have to cascade 2N ports, you can do it without the help of `Circuit` but with the [standard cascading functions](https://scikit-rf.readthedocs.io/en/latest/tutorials/Connecting_Networks.html). These functions should take into account noise data, but not in all cases. Generally speaking, the noise data support could be improved, for example with dedicated documentation. I let specialists of noise data (which I'm not) comment further.

**Comment by Vinc0110** [comment]
Also see the feature request in issue #538. The corresponding PR is still open and unfinished since two years, unfortunately.
